cypress/e2e/invoices/upload-file-to-invoice.cy.js

```diff
@@ -0,0 +1,84 @@
+import Users from '../../support/fragments/users/users';
+import FileManager from '../../support/utils/fileManager';
+
+describe('Invoices', () => {
+  const testData = {
+    invoice: {},
+    organization: NewOrganization.getDefaultOrganization(),
+  };
+  const fileName = `autotestFile.${getRandomPostfix()}.txt`;
+  const linkName = 'Sample link';
+  const linkExample = 'https://example.com';
+
+  before('Create test data and preconditions', () => {
+    cy.getAdminToken();
+    Organizations.createOrganizationViaApi(testData.organization).then(() => {
+      cy.getBatchGroups().then(({ id: batchGroupId, name: batchGroupName }) => {
+        testData.invoice = Invoices.getDefaultInvoice({
+          batchGroupId,
+          batchGroupName,
+          vendorId: testData.organization.id,
+          vendorName: testData.organization.name,
+          accountingCode: testData.organization.erpCode,
+          invoiceDate: DateTools.getCurrentDate(),
+        });
+      });
+    });
+    cy.createTempUser([Permissions.viewEditCreateInvoiceInvoiceLine.gui]).then((user) => {
+      testData.user = user;
+      cy.loginAsAdmin({
+        path: TopMenu.invoicesPath,
+        waiter: Invoices.waitLoading,
+      });
+    });
+  });
+
+  after('Delete test data', () => {
+    Users.deleteViaApi(testData.user.userId);
+    Organizations.deleteOrganizationViaApi(testData.organization.id);
+  });
+
+  it(
+    'C15856 Upload file to invoice (thunderjet) (TaaS)',
+    {
+      tags: [TestTypes.criticalPath, DevTeams.thunderjet],
+    },
+    () => {
+      // Create an invoice
+      const InvoiceEditForm = Invoices.openInvoiceEditForm({ createNew: true });
+      InvoiceEditForm.checkButtonsConditions([
+        { label: 'Cancel', conditions: { disabled: false } },
+        { label: 'Save & close', conditions: { disabled: true } },
+      ]);
+      // Populate required fields and add file
+      InvoiceEditForm.fillInvoiceFields({
+        invoiceDate: testData.invoice.invoiceDate,
+        status: testData.invoice.status,
+        batchGroupName: testData.invoice.batchGroupName,
+        vendorInvoiceNo: testData.invoice.vendorInvoiceNo,
+        vendorName: testData.invoice.vendorName,
+        paymentMethod: testData.invoice.paymentMethod,
+      });
+      InvoiceEditForm.uploadFile('file.txt', fileName);
+      InvoiceEditForm.checkButtonsConditions([
+        { label: 'Save & close', conditions: { disabled: false } },
+        { label: 'Delete', conditions: { disabled: false } },
+      ]);
+      // Add link
+      InvoiceEditForm.addLinkToInvoice(linkName, linkExample);
+      // Click "Save & close" button on "Create vendor invoice" form
+      InvoiceEditForm.clickSaveButton();
+      // Expand "Links & documents" accordion on created invoice pane
+      InvoiceView.verifyLinksDocumentsSection(linkName, linkExample, fileName);
+      InvoiceView.downloadDocument();
+      FileManager.verifyFileIncludes(fileName, [], true);
```

Comment:
Please add in after block delete method for downloaded file from folder.And I don't understand where file.txt?)

Reply:
fixed, removed the part with 'file.txt'